web/app/Http/Controllers/PageController.php

```diff
@@ -4,19 +4,205 @@
 
 namespace Wikijump\Http\Controllers;
 
+use Illuminate\Contracts\Auth\StatefulGuard;
 use Illuminate\Contracts\View\View;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Wikijump\Common\APIError;
 use Wikijump\Helpers\LegacyTools;
+use Wikijump\Services\Deepwell\Models\Category;
+use Wikijump\Services\Deepwell\Models\Page;
+use Wikijump\Services\Deepwell\Models\User;
 use Wikijump\Services\License\License;
 use Wikijump\Services\License\LicenseMapping;
 
 class PageController extends Controller
 {
+    /** Guard used to handle authentication. */
+    private StatefulGuard $guard;
+
+    /**
+     * @param StatefulGuard $guard
+     */
+    public function __construct(StatefulGuard $guard)
+    {
+        $this->guard = $guard;
+    }
+
+    private function resolvePageType(Request $request): ?array
+    {
+        $output_type = (string) $request->query('type', 'none');
+
+        $metadata = false;
+        $wikitext = false;
+        $html = false;
+
+        switch ($output_type) {
+            case 'all':
+                $metadata = true;
+                $wikitext = true;
+                $html = true;
+                break;
+            case 'metadata':
+                $metadata = true;
+                break;
+            case 'metadata-html':
+                $metadata = true;
+                $html = true;
+                break;
+            case 'metadata-wikitext':
+                $metadata = true;
+                $wikitext = true;
+                break;
+            case 'wikitext':
+                $wikitext = true;
+                break;
+            case 'html':
+                $html = true;
+                break;
+            case 'none':
+                break;
+            default:
+                return null;
+        }
+
+        return [
+            'metadata' => $metadata,
+            'wikitext' => $wikitext,
+            'html' => $html,
+        ];
+    }
+
+    /**
+     * @return Page|Response
+     */
+    private function resolvePage(
+        string $path_type,
+        string $path,
+        bool $wikitext = false,
+        bool $html = false
+    ) {
+        $site = LegacyTools::getCurrentSite();
+        if ($site === null) {
+            return apierror(404, APIError::SITE_NOT_FOUND);
+        }
+
+        $site_id = $site->getSiteId();
+
+        $page = null;
+
+        if ($path_type === 'slug') {
+            $page = Page::findSlug($site_id, $path, $wikitext, $html);
+        } elseif ($path_type === 'id') {
+            $page = Page::findId($site_id, (int) $path, $wikitext, $html);
```

Comment:
Huh, I didn't realize you could just cast. I've been using `intval()`, I should probably change that.